src/fmu/dataio/_metadata.py

```diff
@@ -106,15 +105,16 @@ def generate_export_metadata(
     """
 
     objdata = objectdata_provider_factory(obj, dataio)
-    masterdata = dataio.config.get("masterdata")
 
     return schema.InternalObjectMetadata(
         schema_=TypeAdapter(AnyHttpUrl).validate_strings(SCHEMA),  # type: ignore[call-arg]
         version=VERSION,
         source=SOURCE,
         class_=objdata.classname,
         fmu=_get_meta_fmu(fmudata) if fmudata else None,
-        masterdata=_get_meta_masterdata(masterdata) if masterdata else None,
+        masterdata=dataio.config.masterdata
+        if isinstance(dataio.config, GlobalConfiguration)
+        else None,
```

Comment:
```suggestion
        masterdata=(
            dataio.config.masterdata
            if isinstance(dataio.config, GlobalConfiguration)
            else None
        ),
```
Just for readability 😄

Reply:
True 😃